doc(active_record): adding more examples about scope control (#50342)
rails/rails: guides/source/association_basics.md

end
```

> This will work fine, because both the `Supplier` and the `Account` class are defined within the same scope. This is highly beneficial as it allows you to organize your models according to their scope. For example:

Reviewer: Describing the actual scope (`MyApplication::Business`) would help here I think.
We can also move the comment about "folders" here to help the reader notice what to look for.
```suggestion
This will work fine, because both the `Supplier` and the `Account` class are defined within the same scope (`MyApplication::Business`). This organization allows structuring models into folders based on their scope:
```

> ```ruby
> # app/models/business/account.rb
> class Business::Account < ApplicationRecord
>   belongs_to :business
> end
> 
> # app/models/business.rb
> class Business < ApplicationRecord
>   has_many :accounts
> end
> ```

Reviewer: Keeping the same nested examples makes things more consistent.
````suggestion
```ruby
# app/models/my_application/business/supplier.rb
module MyApplication
  module Business
    class Supplier < ApplicationRecord
      has_one :account
    end
  end
end

# app/models/my_application/business/account.rb
module MyApplication
  module Business
    class Account < ApplicationRecord
      belongs_to :supplier
    end
  end
end
```
````

> This organization allows you to structure your models into folders based on their scope. It is crucial to note that this does not affect the naming of your tables. For instance, if there is a `Business::Supplier` model, there must also be a `business_suppliers` table.

Reviewer: We can remove the "folders" comment here.
```suggestion
It is crucial to note that this does not affect the naming of your tables. For instance, if there is a `MyApplication::Business::Supplier` model, there must also be a `my_application_business_suppliers` table.
```

> This organization allows you to structure your models into folders based on their scope. It is crucial to note that this does not affect the naming of your tables. For instance, if there is a `Business::Supplier` model, there must also be a `business_suppliers` table.
> 
> But the following will _not_ work, because `Supplier` and `Account` are defined in different scopes:

Reviewer: The "But" doesn't make much sense here anymore, as it no longer applies to the previous sentence: 
```suggestion
Note that he following will _not_ work, because `Supplier` and `Account` are defined in different scopes (`MyApplication::Business` and `MyApplication::Billing`):
```

end
```

> This will work fine, because both the `Supplier` and the `Account` class are defined within the same scope (`MyApplication::Business`). This organization allows structuring models into folders based on their scope:

Reviewer: A small change to maybe make things a bit clearer:
```suggestion
This will work fine, because both the `Supplier` and the `Account` class are defined within the same scope (`MyApplication::Business`). This organization allows structuring models into folders based on their scope, without having to explicitly add the scope to every association:
```

Author: Done!
